src/components/SelectSlider.tsx

```diff
@@ -0,0 +1,51 @@
+import classNames from "classnames";
+import { ReactNode } from "react";
+import styles from "./SelectSlider.module.scss";
+
+interface SelectProps<T extends string> {
+  name: string;
+  options: Option<T>[];
+  value: T;
+  onChange(value: T): void;
+}
+
+type Option<T extends string> = T | { name: T; node: ReactNode };
```

Comment:
I took quite a bit of time, trying to understand the props.
Some little documentation about what all of these do in conjunction with each other like ->
```ts
value: T; // String identifying the selected element
```
maybe what role  `T` or `name` of `type Option` play in that (might choose a better name than `name` like `value` 🤷).